### PR TITLE
Replace mentions of IRC with Discord

### LINKED ISSUE #61524: Replace mention of IRC with Discord
There are still [few mentions](https://github.com/rust-lang/rust/search?q=IRC&unscoped_q=IRC) of IRC in the repo. Removing them is not a priority but it will avoid confusion for newcomers.

### PR DESCRIPTION
**Pull request by radu-matei**
closes #61524 

One occurrence that requires more attention is in the code of conduct - please make sure it is correctly worded.
Happy to get feedback and reword any other change made.

Thanks!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -419,7 +418,7 @@ Find a [requested community library][community-library], build it, and publish
 it to [Crates.io](http://crates.io). Easier said than done, but very, very
 valuable!
 
-[pound-rust]: http://chat.mibbit.com/?server=irc.mozilla.org&channel=%23rust
+[rust-discord]: http://discord.gg/rust-lang
```
Comment (phansch) [inline CONTRIBUTING.md]:
```suggestion
[rust-discord]: https://discord.gg/rust-lang
```

File: README.md [hunk 2]
```diff
@@ -241,19 +241,17 @@ The Rust community congregates in a few places:
...
 
 The [rustc guide] might be a good place to start if you want to find out how
 various parts of the compiler work.
 
 Also, you may find the [rustdocs for the compiler itself][rustdocs] useful.
 
-[IRC]: https://en.wikipedia.org/wiki/Internet_Relay_Chat
-[#rust]: irc://irc.mozilla.org/rust
-[#rust-beginners]: irc://irc.mozilla.org/rust-beginners
+[rust-discord]: http://discord.gg/rust-lang
```
Comment (phansch) [inline README.md]:
```suggestion
[rust-discord]: https://discord.gg/rust-lang
```

File: src/bootstrap/README.md [hunk 3]
```diff
@@ -328,6 +328,9 @@ are:
   `Config` struct.
 * Adding a sanity check? Take a look at `bootstrap/sanity.rs`.
 
-If you have any questions feel free to reach out on `#rust-infra` on IRC or ask on
+If you have any questions feel free to reach out on `#infra` channel in the 
+[Rust Discord server][rust-discord], or ask on
 internals.rust-lang.org. When you encounter bugs, please file issues on the
 rust-lang/rust issue tracker.
+
+[rust-discord]: http://discord.gg/rust-lang
```
Comment (phansch) [inline src/bootstrap/README.md]:
```suggestion
[rust-discord]: https://discord.gg/rust-lang
```

**Review by phansch — COMMENTED** [review]
Just some nits to use https over http

(I don't have perms to give final approval, though)

**Comment by Manishearth** [comment]
Note: #rust-libs is still active as a team venue (as is #rust-embedded), they're going to find a new place to go before Mozilla IRC shuts down (there's still time).

Furthermore, #rust, #rust-beginners, and #rust-internals are also somewhat active: I plan to lock them at some point to nudge people to Discord/Freenode, but that's not done yet, and the CoC still applies there.

**Comment by Mark-Simulacrum** [comment]
I've added this to the core team agenda to discuss on Wednesday; I think we'll probably want to just _add_ Discord versus replacing IRC with it from the CoC.

**Comment by Manishearth** [comment]
I'm surprised Discord isn't on there already, it is on the website: https://www.rust-lang.org/policies/code-of-conduct

**Comment by Mark-Simulacrum** [comment]
We discussed this a bit during the core team meeting today and decided that we should remove the code of conduct itself from this repository and instead simply link to https://www.rust-lang.org/policies/code-of-conduct. That way there's a canonical version that we can update without having to update numerous repositories. We'll want to edit the website to include IRC as well.

**Comment by steveklabnik** [comment]
One small note about this: we should keep the *files*, but have them say "the text lives at this URL". GitHub will say that the project does not have a code of conduct without the file existing.

**Comment by lzutao** [comment]
@radu-matei you have nits to solve.

**Comment by joelpalmer** [comment]
Hi @radu-matei - this is a ping from Triage. Are there any updates on this?

**Comment by JohnTitor** [comment]
ping from triage @radu-matei
Unfortunately, we haven't heard from you on this in a while, so I'm closing the PR to keep things tidy. Don't worry though, if you'll have time again in the future please reopen this PR, we'll be happy to review it again! Thanks for taking the time to contribute.

**Comment by BO41** [comment]
> We discussed this a bit during the core team meeting today and decided that we should remove the code of conduct itself from this repository and instead simply link to https://www.rust-lang.org/policies/code-of-conduct. That way there's a canonical version that we can update without having to update numerous repositories. We'll want to edit the website to include IRC as well.

@Mark-Simulacrum this never got addressed, right? Is there still an interest?

**Comment by Mark-Simulacrum** [comment]
I think so, yes. I'm happy to review a PR to relevant repositories.

**Pull request closed without merging**